src/core/net/ip6.cpp

```diff
@@ -1230,6 +1231,22 @@ otError Ip6::HandleDatagram(Message &aMessage, Netif *aNetif, const void *aLinkM
             ExitNow(tunnel = true);
         }
 
+        if (nextHeader == kProtoTcp || nextHeader == kProtoUdp)
+        {
+            uint16_t dstPort;
+
+            // TCP/UDP shares header uint16_t srcPort, uint16_t dstPort
+            VerifyOrExit(aMessage.Read(aMessage.GetOffset() + sizeof(uint16_t), sizeof(dstPort), &dstPort) ==
+                             sizeof(dstPort),
+                         error = OT_ERROR_PARSE);
+            dstPort = HostSwap16(dstPort);
+            if (aMessage.IsLinkSecurityEnabled() && Get<ot::Ip6::Filter>().HasUnsecurePort(dstPort))
+            {
+                otLogInfoIp6("Will remove unsecure port %d", dstPort);
```

Comment:
Move to after the line below?
```suggestion
                otLogInfoIp6("Removed unsecure port %u", dstPort);
```